core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -177,6 +177,11 @@ public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData
 
     @Override
     public void onSettingsRead(ChannelHandlerContext ctx, Http2Settings settings) {
+        if (settings.isEmpty()) {
```

Comment:
No, it isn't. However, if the following log is only printed at the debug level, I thought, it could be confusing that users assume the `Http2Settings` is the final negotiated settings.
https://github.com/line/armeria/blob/b127cd27252f6a454130f66d1175a06faed01f09/core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java#L266

Reply:
Fair enough 😄